### PR TITLE
Add a sanity check document

### PR DESCRIPTION
**Pull request by dnadales**
This document is intended for anyone testing or benchmarking the Consensus component outside this repository, for instance `cardano-node` or `cardano-node-tests`.

### CODE REVIEW

File: docs/website/docs/SanityChecks.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+
+## Enable assertions 
+
+Even if the Consensus code is being benchmarked, it might be a good idea to *temporarily* enable assertions to perform additional checks on consistency.
+The following snippet can be added to the `cabal.project` file to enable assertions in our two main components:
+
+```cabal
+package ouroboros-consensus
+  ghc-options: -fno-ignore-asserts
```
Comment (jasagredo) [inline docs/website/docs/SanityChecks.md:13]: Isn't `flags: +asserts` the same?

Comment (jasagredo) [reply docs/website/docs/SanityChecks.md:13]: `program-options` won't work if invoked from the node's `cabal.project` though @jorisdral

**Comment by jasagredo** [comment]
Perhaps we could add a section already speaking about configurations? Something like:

```
- `k`: has to be the same for all eras
- SnapshotInterval: has to be manually set or will be computed as `2*k` slots (care if `k` is small!)
- Mempool size: default is ...
```

File: docs/website/docs/SanityChecks.md (hunk 1 above)
Comment (dnadales) [reply docs/website/docs/SanityChecks.md:13]: Thank you guys, I didn't know that we could use `+asserts` instead.

**Comment by dnadales** [comment]
Thank you for the feedback. I pushed a new commit to address your comments. Please re-check :white_check_mark:

File: docs/website/docs/SanityChecks.md
[the remark this answers is not in the document]
Comment (dnadales) [reply docs/website/docs/SanityChecks.md:1]: That was what I used to do, however @nfrisby made me doubt this rule when it comes to prose: https://github.com/input-output-hk/ouroboros-consensus/pull/11#discussion_r1129766521

I'm happy either way :)

File: docs/website/docs/SanityChecks.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Even if the Consensus code is being benchmarked, it might be a good idea to *temporarily* enable assertions to perform additional checks on consistency.
+The following snippet can be added to the `cabal.project` file to enable assertions in our two main components:
+
+```cabal
+package ouroboros-consensus
+  ghc-options: +asserts
+
+package ouroboros-consensus-cardano
+  ghc-options: +asserts
```
Comment (amesgen) [inline docs/website/docs/SanityChecks.md:16]:
```suggestion
package ouroboros-consensus
  flags: +asserts

package ouroboros-consensus-cardano
  flags: +asserts
```

Comment (dnadales) [reply docs/website/docs/SanityChecks.md:16]: :facepalm: Thanks!

**Review by amesgen — APPROVED** [review]

**Comment by angerman** [comment]
rebased to re-excite CI.

**Pull request merged**